src/com/owncloud/android/lib/resources/users/GetRemoteUserInfoOperation.java

```diff
@@ -140,73 +124,25 @@ protected RemoteOperationResult run(OwnCloudClient client) {
                 String response = get.getResponseBodyAsString();
                 Log_OC.d(TAG, "Successful response: " + response);
 
-                // Parse the response
-                JSONObject respJSON = new JSONObject(response);
-                JSONObject respOCS = respJSON.getJSONObject(NODE_OCS);
-                JSONObject respData = respOCS.getJSONObject(NODE_DATA);
+                Gson gson = new Gson();
+                JsonParser parser = new JsonParser();
+                JsonObject respJson = (JsonObject)parser.parse(response);
+                Type respType = new TypeToken<ServerResponse<UserInfo>>(){}.getType();
+                ServerResponse<UserInfo> ocsResponse = gson.fromJson(respJson, respType);
 
-                UserInfo userInfo = new UserInfo();
+                UserInfo userInfo = ocsResponse.ocs.data;
```

Comment:
Woot! What about using getters instead? getOcs().getData()? :)

Reply:
Both ways for me. not using setters mimics traversing JSON object